### PR TITLE
Implemented Type-Checkable Language Classes

### PR DESCRIPTION
**Pull request by maweki**
Fixing #64 I made all language modules into languages classes.

* I've made some fixed-length lists to tuples
* I've removed one instance of duplicated code
* I fixed typing for all non-empty attributes that can be seen in all other code
* Missing: mypy needs type-hints for empty stuff, which I didn't do yet but is a trivial addition (see errors)
* I've added multiple todos where I feel that a different type is more appropriate, but didn't want to fix that yet.

I've split this into mutliple commits that you probably should view seperately.

```
$ mypy gourmet/defaults/defaults*.py
gourmet/defaults/defaults_sk.py:82: error: Need type annotation for 'SYNONYMS' (hint: "SYNONYMS: List[<type>] = ...")
gourmet/defaults/defaults_sk.py:85: error: Need type annotation for 'AMBIGUOUS' (hint: "AMBIGUOUS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_sk.py:535: error: Need type annotation for 'IGNORE' (hint: "IGNORE: List[<type>] = ...")
gourmet/defaults/defaults_sk.py:537: error: Need type annotation for 'NUMBERS' (hint: "NUMBERS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_pt.py:32: error: Need type annotation for 'SYNONYMS' (hint: "SYNONYMS: List[<type>] = ...")
gourmet/defaults/defaults_pt.py:37: error: Need type annotation for 'AMBIGUOUS' (hint: "AMBIGUOUS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_pt.py:45: error: Need type annotation for 'INGREDIENT_DATA' (hint: "INGREDIENT_DATA: List[<type>] = ...")
gourmet/defaults/defaults_pt.py:193: error: Need type annotation for 'IGNORE' (hint: "IGNORE: List[<type>] = ...")
gourmet/defaults/defaults_pt.py:195: error: Need type annotation for 'NUMBERS' (hint: "NUMBERS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_nl.py:31: error: Need type annotation for 'SYNONYMS' (hint: "SYNONYMS: List[<type>] = ...")
gourmet/defaults/defaults_nl.py:34: error: Need type annotation for 'AMBIGUOUS' (hint: "AMBIGUOUS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_nl.py:437: error: Need type annotation for 'IGNORE' (hint: "IGNORE: List[<type>] = ...")
gourmet/defaults/defaults_fr.py:1138: error: Need type annotation for 'NUMBERS' (hint: "NUMBERS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_es.py:37: error: Need type annotation for 'AMBIGUOUS' (hint: "AMBIGUOUS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_es.py:447: error: Need type annotation for 'IGNORE' (hint: "IGNORE: List[<type>] = ...")
gourmet/defaults/defaults_de.py:120: error: Need type annotation for 'NUTRITIONAL_INFO' (hint: "NUTRITIONAL_INFO: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_de.py:836: error: Need type annotation for 'NUMBERS' (hint: "NUMBERS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_sv.py:67: error: Need type annotation for 'NUTRITIONAL_INFO' (hint: "NUTRITIONAL_INFO: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_sv.py:79: error: Need type annotation for 'AMBIGUOUS' (hint: "AMBIGUOUS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_ru.py:92: error: Need type annotation for 'NUTRITIONAL_INFO' (hint: "NUTRITIONAL_INFO: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_ru.py:104: error: Need type annotation for 'AMBIGUOUS' (hint: "AMBIGUOUS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_en.py:96: error: Need type annotation for 'NUTRITIONAL_INFO' (hint: "NUTRITIONAL_INFO: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_en.py:108: error: Need type annotation for 'AMBIGUOUS' (hint: "AMBIGUOUS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults_en_GB.py:60: error: Need type annotation for 'AMBIGUOUS' (hint: "AMBIGUOUS: Dict[<type>, <type>] = ...")
gourmet/defaults/defaults.py:18: error: Module 'ctypes' has no attribute 'windll'
Found 25 errors in 11 files (checked 11 source files)
```

### CODE REVIEW

File: gourmet/defaults/abstractLang.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+from typing import Any, Mapping, List, Collection, Sequence, Optional, Tuple
+
+class AbstractLanguage:
+
+    ''' Translator credits '''
```
Comment (cydanil) [inline gourmet/defaults/abstractLang.py]: Shouldn't all these triple quotes be `"""` or even block comments?

File: gourmet/defaults/defaults.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import locale, os
+from typing import Optional
+from .abstractLang import AbstractLanguage
 deflang = 'en'
-lang = None
+lang: AbstractLanguage
```
Comment (cydanil) [inline gourmet/defaults/defaults.py:5]: I'm not familiar with this syntax, can you shed some light?

**Review by cydanil — APPROVED** [review]
It's good, although I did not bother to look at each and every default. I have a nitpick and a question for my general understanding.

Other than that, I see the point in making these classes, I find it more harmonious and seem generally more intuitive to use.

File: gourmet/defaults/abstractLang.py (hunk 1 above)
Comment (maweki) [reply gourmet/defaults/abstractLang.py]: Oh, okay. That's an easy enough fix. I always get confused with all the PEPs

File: gourmet/defaults/defaults.py (hunk 2 above)
Comment (maweki) [reply gourmet/defaults/defaults.py:5]: This is just a type definition. If lang is ever `None`, we need an `Optional[Language]` which defeats the purpose. But I haven't noticed, that if all imports fail `if not lang:` gives a `NameError`. I will fix that.

We will probably, at some point, need to change code that is something like `x = x[0]` as this gives a stupid type for `x`.

**Comment by maweki** [comment]
I think I've addressed the issues

**Comment by cydanil** [comment]
Neat! Thanks a lot :)

**Pull request merged**